File: (8 kyu) Count of positives - sum of negatives/(8 kyu) Count of positives - sum of negatives.js
// 1 Plain Solution
function countPositivesSumNegatives(input) {
  if (!input || !input.length) {
    return [];
  }
  const res = [0, 0];
  input.forEach((x) => (x > 0 ? res[0]++ : (res[1] += x)));

  return res;
}
// 2 Optimized solution
function countPositivesSumNegatives(input) {
  return input && input.length
    ? [
        input.filter((p) => p > 0).length,
        input.filter((n) => n < 0).reduce((a, b) => a + b, 0),
      ]
    : [];
}
// 3 Coding golf
const countPositivesSumNegatives = input =>
  input && input.length
    ? input.reduce(
        (p, x) => {
          x > 0 ? p[0]++ : (p[1] += x);
          return p;
        },
        [0, 0],
      )
    : [];

File: (8 kyu) Count of positives - sum of negatives/(8 kyu) Count of positives - sum of negatives.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Kata
{
    public static int[] CountPositivesSumNegatives(int[] input)
    {
        if (input == null || input.Length == 0)
        {
          return new int[0];
        }
      
        int[] res = {0,0};
      
        for (int i = 0; i < input.Length; i++)
        {
            if (input[i] > 0)
            {
                res[0] += 1;
            }
            else
            {
              res[1] += input[i];
            }
        }
      
        return res;
    }
}
